picard/util/tags.py

```diff
@@ -40,186 +41,802 @@
-    'releasetype': N_('Release Type'),
-    'remixer': N_('Remixer'),
-    'replaygain_album_gain': N_('ReplayGain Album Gain'),
-    'replaygain_album_peak': N_('ReplayGain Album Peak'),
-    'replaygain_album_range': N_('ReplayGain Album Range'),
-    'replaygain_reference_loudness': N_('ReplayGain Reference Loudness'),
-    'replaygain_track_gain': N_('ReplayGain Track Gain'),
-    'replaygain_track_peak': N_('ReplayGain Track Peak'),
-    'replaygain_track_range': N_('ReplayGain Track Range'),
-    'script': N_('Script'),
-    'show': N_('Show Name'),
-    'showsort': N_('Show Name Sort Order'),
-    'showmovement': N_('Show Work & Movement'),
-    'subtitle': N_('Subtitle'),
-    'syncedlyrics': N_('Synced Lyrics'),
-    'title': N_('Title'),
-    'titlesort': N_('Title Sort Order'),
-    'totaldiscs': N_('Total Discs'),
-    'totaltracks': N_('Total Tracks'),
-    'tracknumber': N_('Track Number'),
-    'website': N_('Artist Website'),
-    'work': N_('Work'),
-    'writer': N_('Writer'),
-}
-
-PRESERVED_TAGS = (
-    '~bitrate',
-    '~bits_per_sample',
-    '~channels',
-    '~dirname',
-    '~extension',
-    '~filename',
-    '~file_created_timestamp',
-    '~file_modified_timestamp',
-    '~format',
-    '~sample_rate',
-    '~video',
-)
+class TagVar:
+    def __init__(
+        self, name, shortdesc=None, longdesc=None,
+        is_preserved=False, is_hidden=False, is_script_variable=True,
+        is_tag=True, is_calculated=False, is_file_info=False
+    ):
+        """
+        shortdesc: Short description (typically one or two words) in title case that is suitable
+                   for a column header.
+        longdesc: Brief description in sentence case describing the tag/variable.  This should
+                  be similar (within reasonable length constraints) to the description in the Picard User
+                  Guide documentation, and could be used as a tooltip when reviewing a script.
+        is_preserved: the tag is preserved (boolean, default: False)
+        is_hidden: the tag is "hidden", name will be prefixed with "~" (boolean, default: False)
+        is_script_variable: the tag can be used as script variable (boolean, default: True)
+        is_tag: the tag is an actual tag (not a calculated or derived one) (boolean, default: True)
+        is_calculated: the tag is obtained by external calculation (boolean, default: False)
+        is_file_info: the tag is a file information, displayed in file info box (boolean, default: False)
+        """
+        self._name = name
+        self._shortdesc = shortdesc
+        self._longdesc = longdesc
+        self.is_preserved = is_preserved
+        self.is_hidden = is_hidden
+        self.is_script_variable = is_script_variable
+        self.is_tag = is_tag
+        self.is_calculated = is_calculated
+        self.is_file_info = is_file_info
+
+    @property
+    def shortdesc(self):
+        """default to name"""
+        if self._shortdesc:
```

Comment:
I know it should never happen, but do we need to protect against the case where the short description (and/or the long description) has been saved as one or more spaces?  If so, perhaps this could be something like `if self._shortdesc and self._shortdesc.strip():`?

Reply:
Well, I wouldn't bother with that, first because it's unlikely, and second because it wouldn't cause much issues.